source/custom/nirfmxinstr_service.custom.cpp

```diff
@@ -1,2 +1,65 @@
+#include <nirfmxinstr/nirfmxinstr_service.h>
+#include <server/converters.h>
+
+#include <sstream>
+
 namespace nirfmxinstr_grpc {
+const auto kErrorReadBufferTooSmall = -200229;
+const auto kWarningCAPIStringTruncatedToFitBuffer = 200026;
+
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+::grpc::Status NiRFmxInstrService::GetNIRFSASessionArray(::grpc::ServerContext* context, const GetNIRFSASessionArrayRequest* request, GetNIRFSASessionArrayResponse* response)
```

Comment:
Note: I use `CustomCode` pretty rarely. This would be a pretty gnarly bit of codegen:
* The only `init_method` that uses an array size mechanism (`ivi-dance-with-a-twist`).
* Error checking for multiple session names input.
* The only method that doesn't actually call a driver API in it's `init_lambda`.

Adding complex codegen can be a real problem for maintainability.

I searched around and I'm pretty sure that this is the only method in this stile (at least in RFmx).

There are a couple of other reasons to prefer codegen to custom code:
* Better testability (implement in fake service).
* (theoretically) easier to keep up-to-date with other codegen changes and features.

We may decide to codegen this eventually (that's what we did for callbacks in DAQ), or we may leave it as-is (quite a few MI methods still use custom_code and have been fine).

How does this work?

The codgenerator generates "everything exept the service implementation": proto file, library wrapper files, client files. Then the service implementation is added in this cpp file.